lib/BridgeXPathAbstract.php

```diff
@@ -0,0 +1,565 @@
+	 * else.
+	 * @param string $value
+	 * @return string
+	 */
+	protected function formatItemUri($value) {
+		if(strlen($value) === 0) {
+			return '';
+		}
+		if(strpos($value, 'http://') === 0 || strpos($value, 'https://') === 0) {
+			return $value;
+		}
+
+		return urljoin($this->feedUri, $value);
+	}
+
+	/**
+	 * Formats the author of a feed item. Takes extracted raw author and returns it formatted
+	 * as string.
+	 * Can be easily overwritten for in case the value needs to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return string
+	 */
+	protected function formatItemAuthor($value) {
+		return $value;
+	}
+
+	/**
+	 * Formats the timestamp of a feed item. Takes extracted raw timestamp and returns unix
+	 * timestamp as integer.
+	 * Can be easily overwritten for example if a special format has to be expected on the
+	 * source website.
+	 * @param string $value
+	 * @return false|int
+	 */
+	protected function formatItemTimestamp($value) {
+		return strtotime($value);
+	}
+
+	/**
+	 * Formats the enclosures of a feed item. Takes extracted raw enclosures and returns them
+	 * formatted as array.
+	 * Can be easily overwritten for in case the values need to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return array
+	 */
+	protected function formatItemEnclosures($value) {
+		return array($this->cleanImageUrl($value));
+	}
+
+	/**
+	 * Formats the categories of a feed item. Takes extracted raw categories and returns them
+	 * formatted as array.
+	 * Can be easily overwritten for in case the values need to be transformed into something
+	 * else.
+	 * @param string $value
+	 * @return array
+	 */
+	protected function formatItemCategories($value) {
+		return array($value);
+	}
+
+	/**
+	 * @param $imageUrl
+	 * @return string|void
+	 */
+	protected function cleanImageUrl($imageUrl)
+	{
+		$result = preg_match('~(?:http(?:s)?:)?[\/a-zA-Z0-9\-_\.]+\.(?:jpg|gif|png|jpeg|ico){1}~', $imageUrl, $matches);
+		if(1 !== $result) {
+			return;
+		}
+		return urljoin($this->feedUri, $matches[0]);
+	}
+
+	/**
+	 * @param $typedResult
+	 * @return string
+	 */
+	protected function getItemValueOrNodeValue($typedResult)
+	{
+		if($typedResult instanceof DOMNodeList) {
+			$item = $typedResult->item(0);
+			if ($item instanceof DOMElement) {
+				return trim($item->nodeValue);
+			} elseif ($item instanceof DOMAttr) {
+				return trim($item->value);
+			}
+		} elseif(is_string($typedResult) && strlen($typedResult) > 0) {
+			return trim($typedResult);
+		}
+	}
+
+	/**
+	 * Fixes feed encoding by invoking PHP's utf8_decode function on extracted texts.
+	 * Useful in case of "broken" or "weird" characters in the feed where you'd normally
+	 * expect umlauts.
+	 *
+	 * @param $input
+	 * @return string
+	 */
+	protected function fixEncoding($input)
+	{
+		return $this->getParam('fix_encoding') ? utf8_decode($input) : $input;
```

Comment:
I think this should be `utf8_encode()` which would convert iso-8859 to utf8. Right now it's the other way around.